src/sentry/event_manager.py

```diff
@@ -911,42 +921,38 @@ def save(self, project_id, raw=False, assume_normalized=False):
         # into tags (logger, level, environment, transaction).  These are
         # different from legacy attributes which are normalized into tags
         # ahead of time (site, server_name).
-        tags = dict(data.get('tags') or [])
-        tags['level'] = level
+        data.setdefault('tags', [])
```

Comment:
Please use `setdefault_path` here to account for `tags` being explicitly `None`.